### PR TITLE
[XPU] refine code for CrossEntropyWithSoftmaxKernel

### PR DESCRIPTION
**Pull request by lj970926**
### PR Category

Performance Optimization

### PR Types

 Performance

### Description

refactor kernel code and improve performance of xpu CrossEntropyWithSoftmaxKernel:
1. remove useless clip because the xpu api do it internally now
2. transpose the input explicitly to remove redundant transposes in softmax

### CODE REVIEW

**Review by runzhech — COMMENTED** [review]
lgtm

File: paddle/phi/kernels/xpu/cross_entropy_kernel.cc [hunk 1]
```diff
@@ -32,140 +32,115 @@ void CrossEntropyWithSoftmaxKernel(const Context& dev_ctx,
...
-                                      labels.data<int64_t>(),
-                                      labels_int_ptr_l3,
-                                      labels.numel());
+    // 2. soft_cross_entropy only
+    const int* labels_data = nullptr;
+    if (labels.dtype() == phi::DataType::INT32) {
+      labels_data = labels.data<int>();
+    } else if (labels.dtype() == phi::DataType::INT64) {
+      int* labels_tmp = RAII_GUARD.alloc_l3_or_gm<int>(labels.numel());
+      r = xpu::cast<int64_t, int>(dev_ctx.x_context(),
```
Comment (houj04) [inline paddle/phi/kernels/xpu/cross_entropy_kernel.cc:106]: 这里有风险不，因为把int64_t转换成int是会丢失信息的。如果原始输入的labels里面有超过int32范围的值，这里cast了之后就会导致计算结果不对吧。是因为xdnn的hard_cross_entropy的TID只能是int的原因？
原版可能也有类似的问题。

Comment (lj970926) [reply paddle/phi/kernels/xpu/cross_entropy_kernel.cc:106]: 是的，XDNN的hard_cross_entropy只支持int类型，这里的行为和之前是一致的。之前没遇到溢出的问题可能是模型当中通常不会有超过INT_MAX种label类型。

这种直接把int64转成int的方式其他算子也在用（比如embedding_grad），究其原因是XPU硬件对64位类型的性能很差，这种先cast再计算的方式很可能比直接计算int64快。

后续可能考虑先测试下API直接用int64算的性能再考虑这块要不要动。

**Review by houj04 — APPROVED** [review]
